Fairy species (from RealzHS, a beginner caster species)

NormalPerson7 opened the pull request:
Currently this is simply what RealzHS had done, renamed from faerie dragon to fairy. We can make further changes from here.

RealzHS committed:
Faerie Dragon add. Has temp/in-progress player doll tile.

RealzHS committed:
Removed hair/pants from FD tile draw, minor FD stat changes, and updated FD species descriptions to match current attributes.

RealzHS committed:
Ignore dirty submodule

RealzHS committed:
Ignore attempt #2

RealzHS committed:
Submodules updated

RealzHS committed:
Faerie Dragon nerfs: Significant MR nerf, starting Dex nerf, 6 aptitudes nerfed.

RealzHS committed:
Faerie Dragon player tile update.

RealzHS committed:
Faerie Dragon update. Added Statue, Lich, and Dragon form tiles. Also minor balance tweak, and some code formatting cleanup.

RealzHS committed:
Faerie Dragon base tile updated.

RealzHS committed:
FD tiles update. Release candidate?

RealzHS committed:
More FD nerfs - worse starting stat distribution, back to int/dex per 4XL instead of just int, and a couple apts nerfed. Willing to consider harsher nerfs (like -1's to XL, Fighting, and Conjurations apts) after wider playtesting.

RealzHS committed:
Cleanup pass on all formatting/checkwhite errors.

RealzHS committed:
Faerie Dragon tile and transforms updated.

RealzHS committed:
Yet another round of nerfs, stats/apts. Keeping major components in place for now, pending wider playtesting and feedback.

RealzHS committed:
.gitmodules fixed

RealzHS committed:
FD - minor bug fixes and formatting fixes.

RealzHS committed:
Fixed automagic bug when playing as FD

RealzHS committed:
FD update. Movespeed while flying normalized. EV bonus while flying halved (to +10%). No spell hunger. HP to -10%. Some apts adjusted. Descriptions updated to match.

RealzHS committed:
Improved implementation of FD mp cost reduction + cleanup.

RealzHS committed:
Formatting cleanup.

RealzHS committed:
FD - minor balance update.

Starting Int down to 11. Alongside split Int/Dex levelup bonuses and
less total potential/ease of using enhancers, DEs will remain the kings
of outside potential spellpower. Adjusted apts such that there's still
clear guidance towards spell schools with damage spells, but softened
some of the heavily negative apts such that an enterprising player could
potentially be successful pursuing an alternate direction should they so
choose.

NormalPerson7 committed:
Change faerie dragons to fairies

This commit changes almost all references to faerie dragons in the code
to fairies (the only place I haven't changed is the tiles files). We
felt that the species was most definitely not a dragon, due to its low
hp, small size, lack of a breath and a different scales mutation to
others. Simply being a fairy fits this role much much better.

I haven't yet compiled this, so if something goes wrong, it's probably
this huge commit.

NormalPerson7 committed:
Adjust/buff fairy aptitudes

This commit adjusts the aptitudes of fairies as follows:
  - Dodging: +1 to +3
  - Spellcasting: +1 to +2
  - Hexes: 0 to -1  (fairies aren't known for their debilitating power)
  - Summonings: 0 to +1
  - Necromancy: +2 to +1  (equalise Summ and Necr apts)
  - Evocations: -1 to +1
  - Stealth: -3 to -2  (-3 is a little harsh)

Vajrapani commented:
Change the obsolete reference from **faerie_dragon_flight** to **fairy_flight** in player.h,

NormalPerson7 committed:
Fix a couple of missed faerie dragon to fairy conversions

Vajrapani: I have reset your merge commit because it may screw things up
later. I am going to clean up your code/look at changing the way the
shimmering scales mutation before re-committing your other commit, and
the change to the shields aptitude i will commit separately.

NormalPerson7 committed:
Make shields a viable option for fairies

The -3 aptitude is worth training for a buckler sometimes, but
especially given how shield plus magical staff is not possible, and how
they already start with bonus shield, it would have been completely
unreasonable ever to train 21 skill for a regular shield. This commit
makes the aptitude +2, to make this a consideration in some
circumstances.

This may be reverted if the consensus becomes to make them tiny.

NormalPerson7 committed:
Give fairies scales similar to draconians', gaining AC/SH per 3 XLs

This removes MUT_SHIMMERING_SCALES and replaces it with an AC/SH boost
which starts at +3, gaining 1 every third level to end +12 at XL 27.
Thanks Vaj for starting this.

NormalPerson7 commented:
This is now ready to merge for testing. I will squash the commits, then rebase and merge.
